egui/src/widgets/plot/mod.rs

```diff
@@ -115,22 +121,48 @@ impl PlotMemory {
 
 // ----------------------------------------------------------------------------
 
+#[derive(Copy, Clone, PartialEq)]
+struct Cursor {
+    orientation: Orientation,
+    point: f64,
+}
+
+impl Cursor {
+    fn vertical(point: f64) -> Self {
+        Self {
+            orientation: Orientation::Vertical,
+            point,
+        }
+    }
+
+    fn horizontal(point: f64) -> Self {
+        Self {
+            orientation: Orientation::Horizontal,
+            point,
+        }
+    }
+}
+
 /// Defines how multiple plots share the same range for one or both of their axes. Can be added while building
 /// a plot with [`Plot::link_axis`]. Contains an internal state, meaning that this object should be stored by
 /// the user between frames.
 #[derive(Clone, PartialEq)]
 pub struct LinkedAxisGroup {
     pub(crate) link_x: bool,
     pub(crate) link_y: bool,
+    pub(crate) link_cursor: bool,
     pub(crate) bounds: Rc<Cell<Option<PlotBounds>>>,
+    cursors: Rc<RefCell<Vec<(Id, Vec<Cursor>)>>>,
 }
```

Comment:
Reusing the same `LinkedAxisGroup` for the cursor feature has a few downsides. In particular, it is (afaict) not possible to JUST link cursors without also linking scrolling. It also makes it more difficult to refactor one feature without disturbing the other. Is there advantage in coupling them that I'm missing?

---

As an alternative, I would propose a separate `CursorGroup`:

```rust
struct CursorGroup {
    /// with a nice dosctring here
    cursors: Vec<(Id, Vec<Cursor>)> 
}
```

```rust
impl Plot {
    pub fn cursor_group(mut self, group: Rc<Cell<CursorGroup>>) -> Self {
        self.cursor_group = Some(group);
        self
    }
}
```